scripts/test/workflow.sh

```diff
@@ -28,88 +30,89 @@ pip -q install -r python/couler/requirements.txt
-            rm -rf /tmp/sqlflow*
-            return 0
-        else
-            echo "Argo workflow ${WORKFLOW_NAME} ${STATUS}"
-            sleep ${CHECK_INTERVAL_SECS}
-        fi
-    done
-    echo "Run CoulerE2ETest failed, launch argo workflow timeout."
+couler run --mode argo --file /tmp/sqlflow_couler.py > /tmp/sqlflow_argo.yaml
+
+MESSAGE=$(kubectl create -f /tmp/sqlflow_argo.yaml)
+WORKFLOW_NAME=$(echo "$MESSAGE" | cut -d ' ' -f 1 | cut -d '/' -f 2)
+echo "Workflow name: $WORKFLOW_NAME"
+
+TIMEOUT="true"
+for _ in {1..30}; do
+    STATUS=$(kubectl get wf "${WORKFLOW_NAME}" -o jsonpath='{.status.phase}')
+    if [[ "$STATUS" == "Succeeded" ]]; then
+        echo "Argo workflow succeeded."
+        kubectl delete wf "${WORKFLOW_NAME}"
+        rm -rf /tmp/sqlflow*
+        TIMEOUT="false"
+        break
+    else
+        sleep "$CHECK_INTERVAL_SECS"
+    fi
+done
+
+if [[ "$TIMEOUT" == "true" ]]; then
+    echo "Workflow job timeout."
     exit 1
-}
-
-############# Run SQLFLow test with Argo Mode #############
-function test_workflow_e2e() {
-    # start a SQLFlow MySQL Pod with testdata
-    kubectl run mysql --port 3306 \
-	    --env="SQLFLOW_MYSQL_HOST=0.0.0.0" \
-	    --env="SQLFLOW_MYSQL_PORT=3306" \
-	    --image="${SQLFLOW_WORKFLOW_STEP_IMAGE}" \
-	    --command -- bash /start.sh mysql
-    POD=$(kubectl get pod -l run=mysql -o jsonpath="{.items[0].metadata.name}")
-
-    for _ in {1..30}
-    do
-        MYSQL_POD_STATUS=$(kubectl get pod "$POD" -o jsonpath='{.status.phase}')
-        echo "${MYSQL_POD_STATUS}"
-        if [[ "${MYSQL_POD_STATUS}" == "Running" ]]; then
-            echo "SQLFlow MySQL Pod running."
-            MYSQL_POD_IP=$(kubectl get pod "$POD" -o jsonpath='{.status.podIP}')
-            export SQLFLOW_TEST_DATASOURCE="mysql://root:root@tcp(${MYSQL_POD_IP}:3306)/?maxAllowedPacket=0"
-            go generate ./...
-            gotest ./cmd/... -run TestEnd2EndWorkflow -v
-            gotest ./pkg/workflow/argo/... -v
-            return 0
-        else
-            echo "Wait SQLFlow MySQL Pod $POD"
-            sleep ${CHECK_INTERVAL_SECS}
-        fi
-    done
-    echo "Run WorkflowE2ETest failed, launch MySQL Pod times out"
+fi
+
+
+echo "Test access MySQL deployed on Kubernetes ..."
+
+# Start a SQLFlow MySQL Pod with testdata
+kubectl run mysql --port 3306 \
+        --env="SQLFLOW_MYSQL_HOST=0.0.0.0" \
+        --env="SQLFLOW_MYSQL_PORT=3306" \
+        --image="sqlflow:mysql"
+POD=$(kubectl get pod -l run=mysql -o jsonpath="{.items[0].metadata.name}")
+
+TIMEOUT="true"
+for _ in {1..30}; do
+    MYSQL_POD_STATUS=$(kubectl get pod "$POD" -o jsonpath='{.status.phase}')
+    echo "${MYSQL_POD_STATUS}"
+    if [[ "${MYSQL_POD_STATUS}" == "Running" ]]; then
+        MYSQL_POD_IP=$(kubectl get pod "$POD" -o jsonpath='{.status.podIP}')
+        echo "MySQL pod IP: $MYSQL_POD_IP"
+        export SQLFLOW_TEST_DATASOURCE="mysql://root:root@tcp(${MYSQL_POD_IP}:3306)/?maxAllowedPacket=0"
```

Comment:
From the logs of travis-CI:
``` text
create DB failed: failed to ping database: dial tcp 172.17.0.7:3306: connect: connection refused" does not contain "runSQLProgram error: unsupported attribute model.no_exists_param"
```

It seems the MySQL Pod is not ready, maybe we can add `kubectl logs ${POD}` to help debug the failed task?

Reply:
It seems `Running` status can not mean MySQL pod is ready. I tried to use [readiness](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes) to wait until the population has done.